src/component.jl

```diff
@@ -184,29 +214,149 @@ Base.sortperm(c::SharedComponent) = sortperm(c.data)
     end
     return h
 end
+
 ########################################
 #                                      #
 #            Iteration                 #
 #                                      #
 ########################################
-struct EntityIterator{T <: Union{IndicesIterator,Indices,AbstractGroup}}
+struct EntityIterator{T <: Union{IndicesIterator,Indices,AbstractGroup}, TT <: Tuple}
     it::T
+    components::TT
 end
 
-Base.eltype(::EntityIterator) = Entity
+@generated function Base.eltype(::EntityIterator{T, TT}) where {T, TT}
+    t = map(x->Ptr{eltype(x)}, TT.parameters)
+    return :(EntityState{Tuple{$t...}})
```

Comment:
Fix to make this inferrable:

```suggestion
    return :(EntityState{Tuple{$(t...)}})
```